File: 2996. Smallest Missing Integer Greater Than Sequential Prefix Sum.java
class Solution {
    public int missingInteger(int[] nums) {
        int sum = nums[0];
        for(int i = 1; i < nums.length && nums[i] == nums[i - 1] + 1; i++){
            sum += nums[i];
        }

        HashSet<Integer> hs = new HashSet<>();
        for(int j = 0; j < nums.length; j++){
            hs.add(nums[j]);
        }

        while(hs.contains(sum)) sum++;
        return sum;
    }
}

File: 2996. Smallest Missing Integer Greater Than Sequential Prefix Sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int sum = nums[0];
        for (int i = 1; i < nums.size() && nums[i] == nums[i - 1] + 1; i++) {
            sum += nums[i];
        }

        unordered_set<int> hs;
        for (int j = 0; j < nums.size(); j++) {
            hs.insert(nums[j]);
        }

        while (hs.count(sum))
            sum++;

        return sum;
    }
};